### DISCUSSION TITLE
Code Highlight in HTML from DOCX

### DISCUSSION THREAD

**Discussion by rgaiacs** [jgm/pandoc#9492] [Q&A]
Given `mwe.md` as

``````
``` python
if true\:
    print("hello")
```
``````

execute

```
pandoc \
    --from markdown \
    --to html \
    --highlight-style=pygments \
    mwe.md
```

produces

```
<div class="sourceCode" id="cb1"><pre
class="sourceCode python"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> true\:</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">&quot;hello&quot;</span>)</span></code></pre></div>
```

**How can I get the above output from a Microsoft Word file (`.docx`)?**

[mwe.docx](https://github.com/jgm/pandoc/files/14373931/mwe.docx) uses a style named `SourceCode` that Pandoc's DOCX reader converts to `<pre><code></code></pre>`. `pandoc --from docx --to html --highlight-style=pygments mwe.docx` produces

```
<pre><code>if true:    
    print(“Hello”)</code></pre>
```

**Comment by jgm** [comment]
Pandoc will only add highlighting if the source code block is marked with the language as one of the class attributes.
You have this when converting from markdown; the block is marked as `python`.
But that information just doesn't appear anywhere in the docx representation of the code block.

**Reply by rgaiacs** [reply]
Thanks for the reply. My understanding is that the solution is to use a filter to add the class attribute. This solution worked for me.
